### PR TITLE
Issue/8277 site creation site details

### LINKED ISSUE #8277: Site Creation: Site Details view
Add a view for entering Site Title and Site Tagline for the new site.

From the design:
site_details

### PR DESCRIPTION
**Pull request by ScoutHarris**
Fixes #8277 

Needs Review: @nheagy 

**To test:**
Start the Site Creation process. Select a Theme. Verify the Site Details view appears.

**Behavior:**
- The Next button is disabled until a Site Title is entered.
- The Site Title is required; the Tagline is optional.
- Site Title field: the keyboard Next button goes to the Site Tagline field.
- Site Tagline field: the keyboard Done button emulates the Next button action.
- If the Next button is somehow enabled without a title, pressing the Next button will present a FancyAlertViewController error.
- Tapping the view will dismiss the keyboard.
- The text field clear button appears when editing.
- The view height adjusts with the keyboard, resulting in the Next button being repositioned with the keyboard.

Pressing Next with a valid Title will present a temporary confirmation alert:

site_details_confirmation_alert

**Implemented:**

site_details

**Designed:**

site_details_design

======================

site_details

### CODE REVIEW

**Comment by ScoutHarris** [comment]
@folletto - I sent you a video via Slack. Could you please clear up the question about the label text and placeholder text as to what you intended the layout to be? 
As stated by @nheagy above - "I'm thinking that the text field labels and left/right alignment should match the mockup. The style you've used is consistent with the rest of Login, but I assume @folletto has his reasons!"

File: WordPress/Classes/ViewRelated/NUX/SiteDetailsViewController.swift [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+    // MARK: - SigninKeyboardResponder Properties
+
+    @IBOutlet weak var bottomContentConstraint: NSLayoutConstraint?
+    @IBOutlet weak var verticalCenterConstraint: NSLayoutConstraint?
+
+    // MARK: - Properties
+
+    @IBOutlet weak var stepLabel: UILabel!
+    @IBOutlet weak var stepDescrLabel1: UILabel!
```
[the remark this answers is not in the document]
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/NUX/SiteDetailsViewController.swift]: Sorry, you mentioned this before! I just assume 'Descr' is obvious, but I can certainly change it. No prob.

File: WordPress/Classes/ViewRelated/NUX/SigninKeyboardResponder.swift [hunk 2]
```diff
@@ -145,7 +145,7 @@ extension SigninKeyboardResponder where Self: NUXAbstractViewController {
         // If an external keyboard is connected, the ending keyboard frame's maxY
         // will exceed the height of the view controller's view.
         // In these cases, just adjust the height by the amount of the keyboard visible.
-        if keyboardFrame.maxY > view.frame.height {
+        if keyboardFrame.maxY > UIScreen.main.bounds.size.height {
```
[the remark this answers is not in the document]
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/NUX/SigninKeyboardResponder.swift]: I was going to give it a testing (now that I have an external keyboard). But, yea, I'll make a note to have it tested.

File: WordPress/Classes/ViewRelated/NUX/SiteDetailsViewController.swift [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+            siteTitleField.becomeFirstResponder()
+        }
+    }
+
+    private func configureView() {
+        _ = addHelpButtonToNavController()
+
+        navigationItem.title = NSLocalizedString("Create New Site", comment: "Create New Site title.")
+        WPStyleGuide.configureColors(for: view, andTableView: nil)
+        tagDescrLabel.textColor = WPStyleGuide.greyDarken20()
```
[the remark this answers is not in the document]
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/NUX/SiteDetailsViewController.swift]: Good point. Is there a standard/preference? I do like setting that in the storyboard (clearly I missed this one). But I know that's not obvious. I'll roll either way!

**Comment by folletto** [comment]
> "I'm thinking that the text field labels and left/right alignment should match the mockup. The style you've used is consistent with the rest of Login, but I assume @folletto has his reasons!"

Correct. The reasoning for the above is:

1. To differentiate from the login / signup fields as they aren't login details.
2. To align with settings interfaces that instead use that patter (the site title and subtitle are settings).

Note also that the placeholder inside the field becomes invisible when there's content, thus making harder to know what is what once the text is written.

tl;dr: yes please align it with the design blueprints. Thanks!

**Comment by ScoutHarris** [comment]
@nheagy , @folletto - all requested changes have been made. Please review again. Thank you!

**Comment by nheagy** [comment]
Well @ScoutHarris pointed out that the other login fields don't handle dynamic type either. IMO it's just more apparent because of the labels on the left 😉 

I say :shipit: and we can make a cleanup issue to fix all the NUX text fields for dynamic type.

**Comment by folletto** [comment]
> I say :shipit: and we can make a cleanup issue to fix all the NUX text fields for dynamic type.

👍
